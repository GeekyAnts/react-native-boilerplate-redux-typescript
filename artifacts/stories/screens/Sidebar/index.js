import * as React from "react";
import { Text, Container, List, ListItem, Content } from "native-base";
import { NavigationActions } from "react-navigation";
const routes = [
    {
        route: "Home",
        caption: "Home",
    },
    {
        route: "BlankPage",
        caption: "Blank Page",
    },
    {
        route: "Login",
        caption: "Logout",
    },
];
const resetAction = NavigationActions.reset({
    index: 0,
    actions: [NavigationActions.navigate({ routeName: "Login" })],
});
export default class Sidebar extends React.Component {
    render() {
        return (React.createElement(Container, null,
            React.createElement(Content, null,
                React.createElement(List, { style: { marginTop: 40 }, dataArray: routes, renderRow: data => {
                        return (React.createElement(ListItem, { button: true, onPress: () => {
                                data.route === "Login"
                                    ? this.props.navigation.dispatch(resetAction)
                                    : this.props.navigation.navigate(data.route);
                            } },
                            React.createElement(Text, null, data.caption)));
                    } }))));
    }
}
//# sourceMappingURL=index.js.map