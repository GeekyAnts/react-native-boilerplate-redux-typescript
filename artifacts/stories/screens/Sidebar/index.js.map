{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../../src/stories/screens/Sidebar/index.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAC/B,OAAO,EAAE,IAAI,EAAE,SAAS,EAAE,IAAI,EAAE,QAAQ,EAAE,OAAO,EAAE,MAAM,aAAa,CAAC;AACvE,OAAO,EAAE,iBAAiB,EAAE,MAAM,kBAAkB,CAAC;AAErD,MAAM,MAAM,GAAG;IACd;QACC,KAAK,EAAE,MAAM;QACb,OAAO,EAAE,MAAM;KACf;IACD;QACC,KAAK,EAAE,WAAW;QAClB,OAAO,EAAE,YAAY;KACrB;IACD;QACC,KAAK,EAAE,OAAO;QACd,OAAO,EAAE,QAAQ;KACjB;CACD,CAAC;AAMF,MAAM,WAAW,GAAG,iBAAiB,CAAC,KAAK,CAAC;IAC3C,KAAK,EAAE,CAAC;IACR,OAAO,EAAE,CAAC,iBAAiB,CAAC,QAAQ,CAAC,EAAE,SAAS,EAAE,OAAO,EAAE,CAAC,CAAC;CAC7D,CAAC,CAAC;AACH,MAAM,CAAC,OAAO,cAAe,SAAQ,KAAK,CAAC,SAAuB;IACjE,MAAM;QACL,MAAM,CAAC,CACN,oBAAC,SAAS;YACT,oBAAC,OAAO;gBACP,oBAAC,IAAI,IACJ,KAAK,EAAE,EAAE,SAAS,EAAE,EAAE,EAAE,EACxB,SAAS,EAAE,MAAM,EACjB,SAAS,EAAE,IAAI,CAAC,EAAE;wBACjB,MAAM,CAAC,CACN,oBAAC,QAAQ,IACR,MAAM,QACN,OAAO,EAAE,GAAG,EAAE;gCACb,IAAI,CAAC,KAAK,KAAK,OAAO;oCACrB,CAAC,CAAC,IAAI,CAAC,KAAK,CAAC,UAAU,CAAC,QAAQ,CAAC,WAAW,CAAC;oCAC7C,CAAC,CAAC,IAAI,CAAC,KAAK,CAAC,UAAU,CAAC,QAAQ,CAAC,IAAI,CAAC,KAAK,CAAC,CAAC;4BAC/C,CAAC;4BAED,oBAAC,IAAI,QAAE,IAAI,CAAC,OAAO,CAAQ,CACjB,CACX,CAAC;oBACH,CAAC,GACA,CACO,CACC,CACZ,CAAC;IACH,CAAC;CACD"}