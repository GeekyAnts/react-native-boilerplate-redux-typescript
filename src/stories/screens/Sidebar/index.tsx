import * as React from "react";
import { Text, Container, List, ListItem, Content } from "native-base";

const routes = [
  {
    route: "Home",
    caption: "Home"
  },
  {
    route: "BlankPage",
    caption: "Blank Page"
  },
  {
    route: "Login",
    caption: "Logout"
  }
];

export interface Props {
  navigation: any;
}
export interface State {}
export default class Sidebar extends React.Component<Props, State> {
  render() {
    return (
      <Container>
        <Content>
          <List
            style={{ marginTop: 40 }}
            dataArray={routes}
            renderRow={data => {
              return (
                <ListItem
                  button
                  onPress={() => this.props.navigation.navigate(data.route)}
                >
                  <Text>{data.caption}</Text>
                </ListItem>
              );
            }}
          />
        </Content>
      </Container>
    );
  }
}
