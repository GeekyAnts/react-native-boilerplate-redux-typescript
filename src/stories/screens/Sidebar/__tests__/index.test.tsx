import React from "react";
import Sidebar from "../index";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

const navigation = { navigate: jest.fn() };

it("renders correctly", () => {
	const tree = renderer.create(<Sidebar navigation={navigation} />).toJSON();
	expect(tree).toMatchSnapshot();
});
