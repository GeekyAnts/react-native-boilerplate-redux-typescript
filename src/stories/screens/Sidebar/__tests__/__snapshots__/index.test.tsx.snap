// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <RCTScrollView
        closeOnRowBeginSwipe={false}
        closeOnRowPress={true}
        closeOnScroll={true}
        dataArray={
          Array [
            Object {
              "caption": "Home",
              "route": "Home",
            },
            Object {
              "caption": "Blank Page",
              "route": "BlankPage",
            },
            Object {
              "caption": "Logout",
              "route": "Login",
            },
          ]
        }
        dataSource={
          ListViewDataSource {
            "items": 3,
          }
        }
        directionalDistanceChangeThreshold={2}
        disableLeftSwipe={false}
        disableRightSwipe={false}
        enableEmptySections={true}
        leftOpenValue={0}
        previewFirstRow={false}
        recalculateHiddenLayout={false}
        renderRow={[Function]}
        renderScrollComponent={[Function]}
        rightOpenValue={0}
        style={
          Array [
            Object {},
            Object {
              "marginTop": 40,
            },
          ]
        }
        swipeToOpenPercent={50}
        virtual={undefined}
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              button={true}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flexDirection": "row",
                  "marginLeft": 15,
                  "paddingBottom": 13,
                  "paddingRight": 15,
                  "paddingTop": 13,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                Home
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              button={true}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flexDirection": "row",
                  "marginLeft": 15,
                  "paddingBottom": 13,
                  "paddingRight": 15,
                  "paddingTop": 13,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                Blank Page
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              button={true}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flexDirection": "row",
                  "marginLeft": 15,
                  "paddingBottom": 13,
                  "paddingRight": 15,
                  "paddingTop": 13,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                Logout
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
